### PR TITLE
Patch 2919 from Jean-Baptiste

### PR DESCRIPTION
**Pull request by salathe**
https://edit.php.net/?patchID=2919&project=PHP

### CODE REVIEW

File: jbnahan-patch2919.diff [hunk 1]
```diff
@@ -0,0 +1,874 @@
...
++  </methodsynopsis>
++  <para>
++   Change or return the exit code. Used only if the exit code is not gracefuly. If the value is not zero, the recuperation configuration can be used after service fail.
++   See <link xlink:href="https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes">Microsoft system error codes</link> for more details
++  </para>
++  
++  <caution>
++   <para>
++    This function work only in "cli" SAPI. On other SAPI 
++    this function is disabled.
```
Comment (salathe) [inline jbnahan-patch2919.diff]: Is the function disabled (i.e. doesn't appear to exist), or does it return an error (as stated below)?

File: jbnahan-patch2919.diff [hunk 2]
```diff
@@ -0,0 +1,874 @@
...
++     <row>
++      <entry>&Version;</entry>
++      <entry>&Description;</entry>
++     </row>
++    </thead>
++    <tbody>
++     <row>
++      <entry>0.4.0</entry>
++      <entry>
++       This function has been added.
```
Comment (salathe) [inline jbnahan-patch2919.diff]: This change entry is not needed (so the entire changelog section is not needed yet).  The version information for the function should be added to `reference/win32service/versions.xml`

File: jbnahan-patch2919.diff [hunk 3]
```diff
@@ -0,0 +1,874 @@
...
+@@ -79,7 +79,11 @@
+      <term><parameter>Win32ExitCode</parameter></term>
+      <listitem>
+       <para>
+-       If the service exited, the return code from the process.
++       If the service exited, the return code from the process. This value is equal to 
++       <constant>WIN32_ERROR_SERVICE_SPECIFIC_ERROR</constant> 
++       if the exit mode is not gracefuly. See 
++       <link linkend="win32service.constants.errors">Win32Service error codes</link> and 
++       <link linkend="function.win32-set-service-exit-mode"><function>win32_set_service_exit_mode</function></link>
```
Comment (salathe) [inline jbnahan-patch2919.diff]: No need for the `<link>` around the function tags.
`<function>my_function</function>` is enough to link to the documentation for `my_function()`.

File: jbnahan-patch2919.diff (hunk 1 above)
Comment (macintoshplus) [reply jbnahan-patch2919.diff]: No, the function appear to exist but has no effect if used in the code executed outside the cli SAPI.

File: jbnahan-patch2919.diff [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
++
++ <refsect1 role="description"><!-- {{{ -->
++  &reftitle.description;
++  <methodsynopsis role="procedural">
++   <type>int</type><methodname>win32_set_service_exit_code</methodname>
++   <methodparam choice="opt"><type>int</type><parameter>exitCode</parameter><initializer>1</initializer></methodparam>
++  </methodsynopsis>
++  <para>
++   Change or return the exit code. Used only if the exit code is not gracefuly. If the value is not zero, the recuperation configuration can be used after service fail.
++   See <link xlink:href="https://docs.microsoft.com/en-us/windows/desktop/debug/system-error-codes">Microsoft system error codes</link> for more details
```
[the remark this answers is not in the document]
Comment (macintoshplus) [reply jbnahan-patch2919.diff]: Changed for 2 links

File: jbnahan-patch2919.diff (hunk 2 above)
Comment (macintoshplus) [reply jbnahan-patch2919.diff]: For the 3 new functions, I have removed the change log.

File: jbnahan-patch2919.diff (hunk 3 above)
Comment (macintoshplus) [reply jbnahan-patch2919.diff]: Fixed

**Comment by macintoshplus** [comment]
The patch is now updated on edit.php.net.

**Pull request closed without merging**